src/functional/iterator/transforms/pass_manager.py

```diff
@@ -7,41 +9,49 @@
 
 
 def apply_common_transforms(
-    ir,
+    ir: ir.Node,
     use_tmps=False,
     offset_provider=None,
     register_tmp=None,
     unroll_reduce=False,
+    common_subexpression_elimination=True,
 ):
     ir = InlineFundefs().visit(ir)
     ir = PruneUnreferencedFundefs().visit(ir)
     ir = NormalizeShifts().visit(ir)
-    ir = InlineLambdas().visit(ir)
     if not use_tmps:
         for _ in range(10):
             inlined = InlineLifts().visit(ir)
-            inlined = InlineLambdas().visit(inlined)
+            inlined = InlineLambdas.apply(inlined)
             if inlined == ir:
                 break
             ir = inlined
         else:
             raise RuntimeError("Inlining lift and lambdas did not converge.")
+    else:
+        ir = InlineLambdas.apply(ir)
 
     ir = NormalizeShifts().visit(ir)
+
     if unroll_reduce:
         for _ in range(10):
             unrolled = UnrollReduce().visit(ir, offset_provider=offset_provider)
             if unrolled == ir:
                 break
             ir = unrolled
             ir = NormalizeShifts().visit(ir)
-            if not use_tmps:
-                ir = InlineLifts().visit(ir)
+            ir = InlineLifts().visit(ir)
```

Comment:
```suggestion
            if not use_tmps:
                ir = InlineLifts().visit(ir)
```
Why did you remove the check? If it’s not there, the `CreateGlobalTmps` pass below will always be a no-op.

Reply:
Mh, probably a misstake during merging. Fixed.